setup.sh

```diff
@@ -60,3 +60,5 @@ cleanup_project_automation
 rm "$0"
 rm README.md
 mv README_TEMPLATE.md README.md
+
+make generate
```

Comment:
This might break if the user doesn't have `conn-sdk-cli` installed, maybe do `make install-tools` first?

Reply:
Depending on what's in tools.go, it might take a few seconds. To me at least, it's not the best experience, since it's needed only once.

Reply:
Nowhere do we mention this in the [quick start](https://github.com/ConduitIO/conduit-connector-template?tab=readme-ov-file#quick-start), imagine a new user coming to the repo and following the instructions, they will get an error 😕

Reply:
How about this:
```suggestion
if ! command -v conn-sdk-cli >/dev/null 2>&1; then
    echo "conn-sdk-cli is not installed, installing tools needed for connector development"
    make install-tools
fi

make generate
```

Reply:
Adding it to the quick start sounds nice to me. I like the check too, but I think it's simple and good enough to just mention that we have `make install-tools`.

Reply:
Btw, we already mention `make install-tools` in the connector readme: https://github.com/ConduitIO/conduit-connector-template/pull/75/commits/3352c47b38e0e7d4b202135a63a13ecb09e9510a.

Reply:
That's not what I meant. In step 6 it's already too late to install tools x) I'm saying that `make generate` will try to run `conn-sdk-cli specgen` and `conn-sdk-cli readmegen`, so if the tool is not installed at that point you'll see an error. And from the perspective of a new developer that just wants to quickly get started with creating a new connector, that's a really bad experience.

So my suggestion is to not even mention `make install-tools` to the user, just execute it if it's necessary.

Reply:
Hah, I thought we are talking about the Makefile, not about the setup.sh.:) Adding a check here makes sense.